os/net/ipv6/uip6.c

```diff
@@ -2039,21 +2039,22 @@ uip_process(uint8_t flag)
     /* Check the URG flag. If this is set, the segment carries urgent
          data that we must pass to the application. */
     if((UIP_TCP_BUF->flags & TCP_URG) != 0) {
-#if UIP_URGDATA > 0
-      uip_urglen = (UIP_TCP_BUF->urgp[0] << 8) | UIP_TCP_BUF->urgp[1];
-      if(uip_urglen > uip_len) {
+      tmp16 = (UIP_TCP_BUF->urgp[0] << 8) | UIP_TCP_BUF->urgp[1];
+      if(tmp16 > uip_len) {
         /* There is more urgent data in the next segment to come. */
```

Comment:
Could you put a comment saying that we "cap the effective urgent data length at `urg_len`" or something ?  What I read from the code including the original one is that, urgent data continued in following segments is just discarded, although this is the case in the original code only with `UIP_URGDATA > 0` as you found.

Reply:
Done. You are right, the problem was that the old code discarded more data than there was in the segment, so the pointer pointed "before" the received segment in memory. I only noticed because uIP then segfaulted on Linux.